aca_preview/preview.py

```diff
@@ -457,52 +458,59 @@ def check_enough_guide_for_ers(self, n_required=8):
 
         """
         if self.is_ER and len(self.guides) < n_required:
-            self.add_message('critical', f'ER guides stars: only {len(self.guides)} stars')
+            self.add_message('critical', f'ER guide stars: only {len(self.guides)} stars')
 
-    def check_guide_pos_errs(self):
+    def check_pos_err_guide(self, entry):
         """Warn on stars with larger POS_ERR (warning at 1" critical at 2")
 
         """
-        for star in self.guides:
-            agasc_id = star['id']
-            # POS_ERR is in milliarcsecs in the table
-            pos_err = star['POS_ERR'] * 0.001
-            for limit, category in ((2.0, 'critical'),
-                                    (1.0, 'warning')):
-                if pos_err > limit:
-                    self.add_message(
-                        category,
-                        f'Guide star {agasc_id} has POS_ERR {pos_err:.2f}, limit {limit} arcsec')
-                    break
+        if entry['id'] not in self.guides['id']:
+            return
+        star = self.guides[self.guides['id'] == entry['id']][0]
```

Comment:
This is awkward and repeated code, but I didn't see a something natural that made more sense.  I honestly couldn't remember why the guide table seems to have an 'idx' but that 'idx' does not match the catalog table version, so ended up needing to re-match-things-up for each guide star as checked.

Reply:
There is a "natural" right way to do this, with the `get_id()` method: `star = self.guides.get_id(entry['id'])`.  This is really powerful so remember to use this in proseco-related code.  It works for any of the tables including `StarsTable`.

For the `aca.acqs` table at least, the `idx` column is reserved for the index into the `cand_acqs` table.  I don't remember how guide works.

Reply:
Yes.  Thanks!  I'd forgotten about this but knew there was something better.  And then saw in your #18 and fixed.  Though I'm still not sure if it makes sense to loop over the all entries of the star catalog and only pass guide stars to these guide methods or if it makes sense to pass the entries to the methods and let them reject non-guide items.  Either way, I thought having the individual guide tests run on single items made more sense from a somewhat atomic / testing perspective.

Reply:
> Though I'm still not sure if it makes sense to loop over the all entries of the star catalog and only pass guide stars to these guide methods

Yes, I would write a top-level `if` statement in the loop and only pass guide entries to the routines.  That reduces repetition in the check methods themselves.  I actually started writing a comment to this effect but then went with the perfect is the enemy of good mantra.  So your call at this point.

Reply:
And yes, I agree very much with making the tests atomic and work on just a single entry if that is the test.  I was quite happy to see that change.